### ISSUE TITLE
[Regression][iOS] Janky animation while opening keyboard

### ISSUE THREAD

**Issue by delfme** [flutter/flutter#138845]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

This is a popular issue https://github.com/flutter/flutter/issues/120555 that was fixed with hard work via https://github.com/flutter/engine/pull/43463

After upgrading to (Channel master, 3.17.0-11.0.pre.9) I noticed that we have a regression as shown on attached video here
https://github.com/flutter/engine/pull/43463#issuecomment-1817549928
Tested on iphone 13 pro, ios 16.x

If doable, it would be much appreciated if this fix could be worked asap, as well as the android version of it, which seems to be caused by same root problem https://github.com/flutter/flutter/issues/116836

As mentioned, this UI layout is pervasive and seeing a stuttering / unsynchronised keyboard animation makes apps look cheap.

### Expected results

Keyboard animation should be smooth

### Actual results

Keyboard animation is janky

### Code sample

This is a regression, so code is already available here https://github.com/flutter/engine/pull/43463

### Screenshots or Video

https://github.com/flutter/engine/pull/43463#issuecomment-1817549928

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 3.17.0-11.0.pre.9, on macOS 12.6.5 21G531 darwin-arm64, locale en-IT)
[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] Connected device (3 available)
[✓] Network resources

```

</details>

**Comment by luckysmg** [comment]
Are you on iOS16.7?

**Comment by delfme** [comment]
Im on 16.3.1. Is issue reproducible on your device / OS version?

**Comment by huycozy** [comment]
I used sample code at https://github.com/flutter/flutter/issues/116836#issuecomment-1346159530 and can see the issue on `iPhone 14 Pro, iOS 16.4 (emulator)`, `iPhone 7, iOS 15.7.9`. The issue is more clear if we notice keyboard flashes at the bottom of screen when closing.

The issue happens on master channel. On stable channel, it seems to happen as well but not very clear as on master. I wonder which Flutter version the issue didn't happen?


<details>
<summary>Demo (iPhone 7, iOS 15.7.9)</summary>

| master | stable |
| --------------- | --------------- |
<video src="https://github.com/flutter/flutter/assets/104349824/82619f6b-b725-464c-9a04-de2616dfe924"/> | <video src="https://github.com/flutter/flutter/assets/104349824/a6ef2b91-05a5-490e-956d-6fae46f66878"/>

</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.16.0 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (8 hours ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 119.0.6045.123

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.17.0-14.0.pre.13, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.17.0-14.0.pre.13 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf09a8a24f (4 hours ago), 2023-11-21 15:16:08 -0800
    • Engine revision 7cf9d90d59
    • Dart version 3.3.0 (build 3.3.0-152.0.dev)
    • DevTools version 2.30.0-dev.4
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 119.0.6045.159
    ! Error: iPhone has recently restarted. Xcode will continue when iPhone is unlocked. (code -14)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```
</details>

**Comment by delfme** [comment]
Hello @huycozy, can I suggest to also try this sample code for testing? https://github.com/delfme/flutter-native-text-input-repo
Looking at a real bottom text bar UI, gives a better idea of the issue.
Basically, until we fix this regression and https://github.com/flutter/flutter/issues/116836, in Flutter whatever messaging screen (chat, social media, etc) looks cheap, in the sense that React Native (and even Ionic) looks native in comparison.

In regards of your wondering, I dont remember which flutter version I was using at that time, but when https://github.com/flutter/engine/pull/43463 was merged, I tested it throughly on my iphone pro 13 / iOS 16.3.1 + flutter latest master version at that time, and animation was smooth.

To recap the issue cross platform:
- on iOS, it seems to be `setViewportMetrics` called before vsync process callback. 
- on Android, as per @xu-baolin initial discovery, it seems that native view and the flutter view cannot render the frame at the same vsync.

**Comment by huycozy** [comment]
Thanks for providing a sample code. 

I checked it on the latest stable and master channel. The issue doesn't happen every time but it requires triggering the keyboard to open and close many times. When it happens, the AnimatedContainer (the layout above keyboard) will shake a bit. The strange thing is that when the issue occurs there is no jank from Devtools's Performance page

Do you also notice the same behavior? (I need to confirm this because I'm not sure if we see the same thing. It would be appreciated if you could also provide a good case when the issue didn't happen).

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.16.0, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.16.0 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db7ef5bf9f (8 hours ago), 2023-11-15 11:25:44 -0800
    • Engine revision 74d16627b9
    • Dart version 3.2.0
    • DevTools version 2.28.2

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 119.0.6045.123

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

```bash
[!] Flutter (Channel master, 3.17.0-15.0.pre.13, on macOS 14.1 23B74 darwin-x64, locale en-VN)
    • Flutter version 3.17.0-15.0.pre.13 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 106667eb4b (5 hours ago), 2023-11-22 22:15:57 +0000
    • Engine revision f331e0afaf
    • Dart version 3.3.0 (build 3.3.0-152.0.dev)
    • DevTools version 2.30.0-dev.4
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.8+0-17.0.8b1000.22-10799086)

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe Patch 3.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app/
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.84.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.76.0

[✓] Connected device (4 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D                         • android-arm64  • Android 11 (API 30)
    • iPhone (mobile)  • d9a94afe2b649fef56ba0bfeb052f0f2a7dae95e • ios            • iOS 15.7.9 19H365
    • macOS (desktop)  • macos                                    • darwin-x64     • macOS 14.1 23B74 darwin-x64
    • Chrome (web)     • chrome                                   • web-javascript • Google Chrome 119.0.6045.159

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```
</details>

**Comment by delfme** [comment]
@huycozy my bad, I forgot I was testing AnimatedContainer as a potential workaround for another issue.
Can you please test the sample again?

https://github.com/flutter/flutter/assets/53510751/4ffb6a92-19f9-40f1-9aa8-9a322e79d608

In my case I see:
1) a delay that doesn't match the native look&feel, in the sense that the Container is animated a bit too after the keyboard animation begins and the keyboard hide it for a while. This is maybe inevitable in a cross platform scenario, but I'm wondering if it could be improved.
2) a little shake occurs from time to time. This is ofc the real issue, but it seems that with this basic sample it occurs way more sporadically and it is barely noticeable. @luckysmg might it be that an heavy scene can impact the smoothness of the animation? Coz in our real case scenario, we have a chat messageList with texts and images and they move along with the bottom inset.

**Comment by huycozy** [comment]
I rechecked your updated example, but the problem isn't really clear. Perhaps there is only one obvious problem: the **first time** the keyboard trigger is opened, the Container will be delayed and display slowly after a while (as you mentioned in the first point?). Then it looks fine afterward. (I checked this on both Flutter stable and master channel) 

<details>
<summary>Demo (iPhone 7, iOS 15.7.9 - profile mode)</summary>


https://github.com/flutter/flutter/assets/104349824/81533a10-15da-4fab-92f5-45c226f87f87


</details>

>a little shake occurs from time to time.

I don't experience this. It's quite hard to see in your recorded video, and I didn't notice it on my device either. This is to determine whether the issue occurs on a specific iOS version or device.

It would be appreciated if there is a way to make this more clear.

**Comment by delfme** [comment]
Yea will check again point 2, coz we have the shake inside our app but not here in this sample code. One idea might be that shake could be due to rebuilds. I recall there was an issue of multiple rebuilds when opening keyboard, caused by MediaQuery, maybe we are hit by some rebuilds that make animation shake. Will investigate this and revert.

**Comment by delfme** [comment]
> I rechecked your updated example, but the problem isn't really clear. Perhaps there is only one obvious problem: the **first time** the keyboard trigger is opened, the Container will be delayed and display slowly after a while (as you mentioned in the first point?). Then it looks fine afterward. (I checked this on both Flutter stable and master channel)
> 
> Demo (iPhone 7, iOS 15.7.9 - profile mode)
>  RPReplay_Final1700799120.mp4 
> 
> > a little shake occurs from time to time.
> 
> I don't experience this. It's quite hard to see in your recorded video, and I didn't notice it on my device either. This is to determine whether the issue occurs on a specific iOS version or device.
> 
> It would be appreciated if there is a way to make this more clear.

@huycozy we were unaware of the huge delay you have shown in your video sample, which is ofc super severe bug :) I mean in that video the textfiled is hidden by keyboard for almost 4 sec! @luckysmg does it mean we must tailor the fix accordingly to the iphone model or set of models?

**Comment by delfme** [comment]
I checked again point 2.
It is not due to rebuilds, but to platform view. We use a native textfield inside our app.
Adding it to sample code, so you guys can test it. Now we have two textfield (one is flutter and one is native). If the native one is used, or whatever platform view element is placed over the scene, then the View will shake during the keyboard animation.
https://github.com/delfme/flutter-native-text-input-repo/blob/main/lib/demo_screen.dart

https://github.com/flutter/engine/pull/43463 was working also when a platform view were on the scene. Maybe since then something has changed about platform-view or Impeller.

Shake issue:

https://github.com/flutter/flutter/assets/53510751/99706dc4-7c39-4c3a-b981-93d50956cb97


https://github.com/flutter/flutter/assets/53510751/4e7dbc75-114d-4ee9-bd7d-c7a0cfb7bb97


Timeline
https://drive.google.com/file/d/1kv0m_hApz8Ty8n2Vs-Zc3O2l2UWxaxgi/view?usp=sharing

**Comment by delfme** [comment]
Note: this fix is important not only for those who are using a native textfield inside some specific screens( where the native look&feel is necessary), but also for those who are using flutter textfield, coz the shake issue will occur even with flutter textfield if a platform view is placed on the scene.

**Comment by delfme** [comment]
Note: Issue is reproducible with both Impeller and Skia.

**Comment by delfme** [comment]
It seems that styling the native textfield is what causes the shake issue. It shouldn't be so, coz styling commands are run when View is initialized, but in any event this is something we can workaround. 

Can someone investigate issue point 1? The animation delay doesn't depend on platform view and is pretty bad on older iphone models, see video from this comment https://github.com/flutter/flutter/issues/138845#issuecomment-1825131443

**Comment by delfme** [comment]
@huycozy should we change title to "TextField animation out of sync on older iphone models?"

**Comment by huycozy** [comment]
>It is not due to rebuilds, but to platform view. We use a native textfield inside our app.

>It seems that styling the native textfield is what causes the shake issue. It shouldn't be so, coz styling commands are run when View is initialized, but in any event this is something we can workaround.

I think you should specify which attribute causes the issue. This reminds me to an old issue at https://github.com/flutter/flutter/issues/120434. We should determine this is a package or framework issue.

>we were unaware of the huge delay you have shown in your video sample, which is ofc super severe bug :) I mean in that video the textfiled is hidden by keyboard for almost 4 sec!

I think it's another issue. You can decide if it should be another issue and file one; or keep it as the main issue here. Issues should be segregated for better tracking and investigation (I think you know this well :) )

**Comment by delfme** [comment]
Yes, added it here https://github.com/flutter/flutter/issues/139054
